panics.go

```diff
@@ -448,12 +459,14 @@ func handlePanic(e *Error, timeout time.Duration) {
 		_, _ = e.WriteTo(wr) // ignore error
 	}(wr, done)
 
-	to := time.NewTimer(timeout)
-	select {
-	case <-to.C:
-		// Timed out waiting for write to complete.
-	case <-done:
-		to.Stop()
+	if timeout > 0 {
+		to := time.NewTimer(timeout)
+		select {
+		case <-to.C:
+			// Timed out waiting for write to complete.
+		case <-done:
+			to.Stop()
+		}
```

Comment:
If `timeout <= 0` then we immediately exit and process the panic without waiting for the write to complete. We want the opposite here and should wait for the write to complete before the panic is handled (note: that handling is done in a deferred function).

Reply:
One thought is to skip all the goroutine and channel setup if the `timeout <= 0` and just call `WriteTo` directly:

```go
	if timeout <= 0 {
		func() {
			defer func() { _ = recover() }() // ignore panic
			_, _ = e.WriteTo(wr)
		}()
		return
	}
```

Reply:
I see the first line says `defer process(e)`. Sneaky, missed it, thanks for pointing that out. I was trying to correlate the timer with `e.WriteTo` and was thinking about how there's no way the timer ticking would abort the write. Changed.